frontend/src/utils/appUrl.test.ts

```
@@ -0,0 +1,163 @@
+    MAP_ROOT,
+    { ...MAP_ROOT, mode: 'discover' },
+    { ...MAP_ROOT, worldViewId: 5 },
+    { ...MAP_ROOT, worldViewId: 5, regionId: 6737 },
+    { ...MAP_ROOT, worldViewId: 5, regionId: 6737, experienceId: 1234 },
+    { ...MAP_ROOT, mode: 'discover', worldViewId: 5, regionId: 7120 },
+    { ...MAP_ROOT, mode: 'discover', worldViewId: 5, regionId: 7120, categoryId: 1 },
+    { ...MAP_ROOT, mode: 'discover', worldViewId: 5, regionId: 7120, experienceId: 1234, categoryId: 1 },
+  ];
+
+  it.each(addresses)('%j', (address) => {
+    const url = new URL(buildAppUrl(address, { region: 'Europe', experience: 'Stonehenge' }), 'http://x');
```

ESLint

[error] Using http protocol is insecure. Use https instead.

(sonarjs/no-clear-text-protocols)